api/lib/domain/usecases/reconcile-oidc-user-for-admin.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @typedef {import ('./index.js').OidcAuthenticationService} OidcAuthenticationService
+ * @typedef {import ('./index.js').AuthenticationSessionService} AuthenticationSessionService
+ * @typedef {import ('./index.js').AuthenticationMethodRepository} AuthenticationMethodRepository
+ * @typedef {import ('./index.js').UserRepository} UserRepository
+ * @typedef {import ('./index.js').UserLoginRepository} UserLoginRepository
+ */
+
+import { AuthenticationKeyExpired, DifferentExternalIdentifierError } from '../errors.js';
+import { AuthenticationMethod } from '../models/index.js';
+
+/**
+ * @param {Object} params
+ * @param {string} params.authenticationKey
+ * @param {string} params.email
+ * @param {string} params.identityProvider
+ * @param {OidcAuthenticationService} params.oidcAuthenticationService
+ * @param {AuthenticationSessionService} params.authenticationSessionService
+ * @param {AuthenticationMethodRepository} params.authenticationMethodRepository
+ * @param {UserRepository} params.userRepository
+ * @param {UserLoginRepository} params.userLoginRepository
+ */
+const reconcileOidcUserForAdmin = async function ({
+  authenticationKey,
+  email,
+  identityProvider,
+  oidcAuthenticationService,
+  authenticationSessionService,
+  authenticationMethodRepository,
+  userRepository,
+  userLoginRepository,
+}) {
+  const sessionContentAndUserInfo = await authenticationSessionService.getByKey(authenticationKey);
+  if (!sessionContentAndUserInfo) {
+    throw new AuthenticationKeyExpired();
+  }
+
+  const foundUser = await userRepository.getByEmail(email);
+
+  const authenticationMethods = await authenticationMethodRepository.findByUserId({ userId: foundUser.id });
+  const oidcAuthenticationMethod = authenticationMethods.find(
+    (authenticationMethod) => authenticationMethod.identityProvider === identityProvider,
+  );
+
+  const isSameExternalIdentifier =
+    oidcAuthenticationMethod?.externalIdentifier === sessionContentAndUserInfo.userInfo.externalIdentityId;
+  if (oidcAuthenticationMethod && !isSameExternalIdentifier) {
+    throw new DifferentExternalIdentifierError();
+  }
```

Comment:
Proposition de simplification (pour éviter la double condition).
Je propose aussi de sortir cela dans une fonction à part, par exemple  `_assertExternalIdentifier` (je trouve que ça fait beaucoup de lignes de code dans la fonction principale pour gérer un cas d'erreur).

```suggestion
const isSameExternalIdentifier =
    authenticationMethods
      .filter((authenticationMethod) => authenticationMethod.identityProvider === identityProvider)
      .map((authMethod) => authMethod.externalIdentifier === sessionContentAndUserInfo.userInfo.externalIdentityId) ||
    false;

  if (!isSameExternalIdentifier) {
    throw new DifferentExternalIdentifierError();
  }
```

Reply:
dans ta proposition ça va retourner un tableau non ?